heal-remedy.lic

```diff
@@ -0,0 +1,93 @@
+# quiet
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#tendme
+=end
+
+custom_require.call(%w[common common-healing common-items])
+
+
+arg_definitions = [
+  [
+    { name: 'debug', regex: /debug/i, optional: true }
+  ]
+]
+
+args = parse_args(arg_definitions)
+$debug_mode_hr = args.debug
+
+$remedies = get_data('remedies')
+
+def wounds
+  wounds = DRCH.check_health
+  pause 1
+#  echo (wounds)
+  hurt = wounds.select{|level, _| level >= 4}.values.flatten
+  wound_grouping = { "right arm" => "limbs", "left arm" => "limbs", "left leg" => "limbs", "right leg" => "limbs", "right hand" => "limbs", "left hand" => "limbs", "right eye" => "eyes", "left eye" => "eyes" }
+  scar_grouping = { "head" => "face", "neck" => "face", "eyes" => "face", "chest" => "body", "abdomen" => "body", "back" => "body" }
+  hurt = hurt.map{ |raw| wound_grouping[raw] || raw}.uniq
+  scar = hurt.map{ |raw| scar_grouping[raw] || raw}.uniq
+  echo ("Scars: ") if $debug_mode_hr
+  if $debug_mode_hr
+    scar.each do |key| 
+      puts key
+    end
+  end
+  hurt.each do |key|
+    puts key if $debug_mode_hr
+    echo("In break loop") if $debug_mode_hr
+    apply_wounds(key)
+  end
+  pause 180 # pause 3 minutes before using scar remedies
+  scar.each do |key|
+    puts key if $debug_mode_hr
+    echo("In break loop") if $debug_mode_hr
+    apply_scars(key)
+  end
+end
+  
+def apply_wounds(key)
+  echo ("In apply_wounds function and key is #{key}") if $debug_mode_hr
+  rem_wounds = $remedies['remedies']['external'][key]
+  echo (rem_wounds) if $debug_mode_hr
+  rem_wounds.each do |key|
+    echo("In apply loop") if $debug_mode_hr
+    if DRCI.exists?("#{key}") 
+      fput("get #{key}")
```

Comment:
Could this use `DRC.bput`?

And the same question for each other `fput`...

Reply:
No round time is given for getting, applying, and stowing the remedies. Not sure what bput would offer for this.

Reply:
Roundtime isn't really related to the use of bput, it's more about asserting state. Even when you don't expect roundtime, this is a healing script for when people are injured so being able to gracefully deal with RT from fleas, lag, boomer traps, and player intervention.

Reply:
Okay, changed to DRC.bput instead of fput.

Reply:
To get the benefit of bput, you'll want to add parameters for acceptable output text: `DRC.bput("get #{key}", 'foo', 'bar')`.

Reply:
Added.